controller/array_action/array_mediator_svc.py

```diff
@@ -468,26 +475,16 @@ def _stop_and_delete_fcmap(self, fcmap_id):
         self._stop_fcmap(fcmap_id)
         self._delete_fcmap(fcmap_id, force=True)
 
-    def _delete_all_fcmaps_as_source_if_exist(self, snapshot_name):
-        snapshot_as_source_fcmaps = self._get_fcmaps(snapshot_name, ENDPOINT_TYPE_SOURCE)
-        unfinished_fcmaps = [fcmap.name for fcmap in snapshot_as_source_fcmaps
-                             if fcmap.status != FCMAP_STATUS_DONE]
-        if unfinished_fcmaps:
-            raise controller_errors.ObjectIsStillInUseError(snapshot_name, unfinished_fcmaps)
-        for fcmap in snapshot_as_source_fcmaps:
-            self._delete_fcmap(fcmap.id, force=False)
-
-    def _delete_snapshot(self, cli_volume):
-        snapshot_name = cli_volume.name
-        fcmap = self._get_fcmap_as_target_if_exists(snapshot_name)
-        if not fcmap:
-            raise controller_errors.ObjectNotFoundError(snapshot_name)
-
-        fcmap_id = cli_volume.FC_id
-        if fcmap_id == 'many':
-            self._delete_all_fcmaps_as_source_if_exist(snapshot_name)
-        self._stop_and_delete_fcmap(fcmap.id)
-        self._delete_volume_by_name(snapshot_name)
+    def _delete_object(self, cli_object, snapshot=False):
+        object_name = cli_object.name
+        fcmap_as_target = self._get_fcmap_as_target_if_exists(object_name)
+        if snapshot and not fcmap_as_target:
+            raise controller_errors.ObjectNotFoundError(object_name)
+        if cli_object.FC_id == 'many':
+            self._safe_delete_fcmaps_as_source(object_name)
```

Comment:
this looks very similar to the original `_delete_all_fcmaps_as_source_if_exist`.
so why the rename?

Reply:
why are you care so much bout `_delete_all_fcmaps_as_source_if_exist`? It didnt have what I wanted so I removed it, yes the new method is very similar, but to now look back how it called back then just dont seems very important right now..

Reply:
not sure what you mean by "removed".
a PR is reviewed by its diffs, and each diff should have some motivation behind it.

in this diff, it just seems like the function implementation has been slightly modified (still trying to understand why, in other conversations), and renamed (still trying to understand why, in this conversation).

could we change the subject of this conversation back to the question: why should this function be merged to dev with this new name?

if the reason is something like "I removed it locally and forgot the old name, so I just made up a new one", then please keep the original name.

if the reason is something like "the new name reflects the changes I made in it, and this is why...", then please share.

Reply:
the new name is IMO better regardless of the change, we not necessarily deleting any fcmap so "safe" is needed there